.github/workflows/docs.yml

```diff
@@ -4,23 +4,51 @@ on:
   push:
     branches:
       - main
+      - feature_publish-mkdocs
 
 permissions:
   contents: write
+
 jobs:
   deploy-docs:
     runs-on: ubuntu-latest
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure Git author
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
 
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
-          python-version: 3.11 
+          python-version: "3.11"
+
+      - name: Install system dependencies for dlib
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y cmake build-essential
+
+      - name: Install Python dependencies
+        run: |
+          pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Expose src on PYTHONPATH
+        run: echo "PYTHONPATH=$GITHUB_WORKSPACE/src:$PYTHONPATH" >> $GITHUB_ENV
 
-      - name: Install dependencies
-        run: pip install mkdocs mkdocs-material mike
-      - name: Deploy documentation
-        run: mkdocs gh-deploy --force
\ No newline at end of file
+      - name: Deploy documentation with mike
+        run: |
+          export PATH="$HOME/.local/bin:$PATH"
+          if [ "${{ github.ref_name }}" = "main" ]; then
+            echo "Deploying 'main' branch as 'latest'..."
+            mike deploy --push latest
+          else
+            echo "Deploying branch ${{ github.ref_name }} as ${{ github.ref_name }}..."
+            mike deploy --push "${{ github.ref_name }}"
+          fi
```

Comment:
Here I think the dropdown for different versions/branches would be created but right now it does not work. Eiterh you fix it or you create a new issue for a future Version ;-)